modules/core_project_factory/main.tf

```diff
@@ -283,7 +283,7 @@ resource "google_service_account_iam_member" "service_account_grant_to_group" {
   Account on shared VPC
  *****************************************************************************************************************/
 resource "google_project_iam_member" "controlling_group_vpc_membership" {
-  count = var.shared_vpc != "" && length(compact(var.shared_vpc_subnets)) == 0 ? local.shared_vpc_users_length : 0
+  count = var.shared_vpc_subnets_count != null && length(compact(var.shared_vpc_subnets)) == 0 ? local.shared_vpc_users_length : 0
```

Comment:
same for length(compact(var.shared_vpc_subnets)) , we can define value of this expression it as a local variale and just use it?

Reply:
Better use the boolean variable from your previous comment if it will work